extensions/resource-deployment/src/services/notebookService.ts

```diff
@@ -25,28 +45,66 @@ export class NotebookService implements INotebookService {
 	 * @param notebook the path of the notebook
 	 */
 	launchNotebook(notebook: string | NotebookInfo): Thenable<azdata.nb.NotebookEditor> {
-		const notebookPath = this.getNotebook(notebook);
-		const notebookFullPath = path.join(this.extensionPath, notebookPath);
-		return this.platformService.fileExists(notebookPath).then((notebookPathExists) => {
-			if (notebookPathExists) {
-				return this.showNotebookAsUntitled(notebookPath);
-			} else {
-				return this.platformService.fileExists(notebookFullPath).then(notebookFullPathExists => {
-					if (notebookFullPathExists) {
-						return this.showNotebookAsUntitled(notebookFullPath);
-					} else {
-						throw localize('resourceDeployment.notebookNotFound', "The notebook {0} does not exist", notebookPath);
-					}
-				});
-			}
+		return this.getNotebookFullPath(notebook).then(notebookPath => {
```

Comment:
nit: Suggest making this async and return a promise just as general cleanup

Reply:
This method relies on an extensibility API which returns Thenable.

Reply:
I'd just leave it as is.